server/api/diagnose.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/pd/server"
+	"github.com/unrolled/render"
+	"net/http"
+	"time"
+)
+
+type diagnoseType int
+type diagnoseKV map[diagnoseType]Recommended
+
+//Recommended return detail
+type Recommended struct {
+	Module      string `json:"module"`
+	Level       string `json:"level"`
+	Description string `json:"description"`
+	Instruction string `json:"instruction"`
+}
+
+const (
+	//analyze level
+	levelNormal   = "Normal"
+	levelWarning  = "Warning"
+	levelMinor    = "Minor"
+	levelMajor    = "Major"
+	levelCritical = "Critical"
+	levelFatal    = "Fatal"
+
+	//analyze module
+	modMember   = "member"
+	modTiKV     = "TiKV"
+	modReplica  = "Replic"
+	modSchedule = "Schedule"
+	modDefault  = "indefinite"
+
+	memberOneInstance diagnoseType = iota
+	memberEvenInstance
+	memberLostPeers
+	memberLostPeersMoreThanHalf
+	memberLeaderChanged
+	tikvCap70
+	tikvCap80
+	tikvCap90
+	tikvLostPeers
+	tikvLostPeersLongTime
+)
+
+var (
+	diagnoseMap = diagnoseKV{
```

Comment:
We can just `diagnoseMap = map[diagnoseType]Recommended {...}`. No need to introduce a new type.